stingray/crosscorrelation.py

```diff
@@ -0,0 +1,190 @@
+                 
+         n: int
+             Number of points in self.corr(Length of cross-correlation data) 
+        """
+
+        ## Populate all attributes by None if user passes no lightcurve data
+        if lc1 is None and lc2 is None:
+            self.corr = None
+            self.time_shift = None
+            self.time_lags = None
+            self.dt = None
+            self.n = None
+            return
+        else:
+            self._make_corr(lc1, lc2)
+
+    def _make_corr(self, lc1, lc2):
+
+        """
+        Creates Crosscorrelation Object.
+
+        Parameters
+        ----------
+        lc1: lightcurve.Lightcurve object
+            The first light curve data.
+
+        lc2: lightcurve.Lightcurve object
+            The second light curve data.
+
+        Returns
+        ----------
+        None
+        """
+
+        if not isinstance(lc1, lightcurve.Lightcurve):
+            raise TypeError("lc1 must be a lightcurve.Lightcurve object")
+        if not isinstance(lc2, lightcurve.Lightcurve):
+            raise TypeError("lc2 must be a lightcurve.Lightcurve object")
+
+        # Sizes of both light curves are assumed to be equal for now
+        if lc1.n != lc2.n:
+            raise StingrayError('Both lightcurves should be of same length')
+
+        if not np.isclose(lc1.dt, lc2.dt):
+            raise StingrayError("Light curves do not have "
+                                "same time binning dt.")
+        else:
+            self.dt = lc1.dt
+
+        # Calculates cross-correlation of two lightcurves
+        self.corr = signal.correlate(lc1.counts, lc2.counts)
+        self.n = len(self.corr)
+
+        self.time_shift, self.time_lags, self.n = self.cal_timeshift(dt=self.dt)
+
+    def cal_timeshift(self, dt=1.0):
+        """
+        Creates Crosscorrelation Object.
+
+        Parameters
+        ----------
+        dt: float , optional, default 1.0
+            Time resolution of lightcurve, should be passed when object is populated with correlation data
+            and no information about light curve can be extracted. Used to calculate time_lags.
+
+        Returns
+        ----------
+        self.time_shift: float
+             Value of time lag that gives maximum value of correlation between two lightcurves. 
+
+        self.time_lags: numpy.ndarray
+             An array of time_lags calculated from correlation data 
+        """
+
+        if self.dt is None:
+            self.dt = dt
+
+        if self.corr is None:
+            raise StingrayError('Time Shift cannot be calculated without correlation data')
+
+        self.n = len(self.corr)
+        dur = int(self.n / 2)
+        # Correlation against all possible lags, positive as well as negative lags are stored
+        x_lags = np.linspace(-dur, dur, self.n)
+        self.time_lags = x_lags * self.dt
+        # time_shift is the time lag for max. correlation
+        self.time_shift = self.time_lags[np.argmax(self.corr)]
```

Comment:
Random question (also for @matteobachetti, sorry): is this the same time lag we're calculating from the phases of the cross spectrum or am I confusing something here?